components/skus/renderer/brave_skus_js_handler.cc

```diff
@@ -0,0 +1,352 @@
+  v8::Local<v8::Context> context = isolate->GetCurrentContext();
+  javascript_object
+      ->Set(context, gin::StringToSymbol(isolate, name),
+            gin::CreateFunctionTemplate(isolate, callback)
+                ->GetFunction(context)
+                .ToLocalChecked())
+      .Check();
+}
+
+// window.brave.skus.refresh_order
+v8::Local<v8::Promise> BraveSkusJSHandler::RefreshOrder(v8::Isolate* isolate,
+                                                        std::string order_id) {
+  if (!EnsureConnected())
+    return v8::Local<v8::Promise>();
+
+  v8::MaybeLocal<v8::Promise::Resolver> resolver =
+      v8::Promise::Resolver::New(isolate->GetCurrentContext());
+  if (resolver.IsEmpty()) {
+    return v8::Local<v8::Promise>();
+  }
+
+  auto promise_resolver(
+      v8::Global<v8::Promise::Resolver>(isolate, resolver.ToLocalChecked()));
+  auto context_old(
+      v8::Global<v8::Context>(isolate, isolate->GetCurrentContext()));
+
+  skus_sdk_->RefreshOrder(
+      order_id,
+      base::BindOnce(&BraveSkusJSHandler::OnRefreshOrder,
+                     base::Unretained(this), std::move(promise_resolver),
+                     isolate, std::move(context_old)));
+
+  return resolver.ToLocalChecked()->GetPromise();
+}
+
+void BraveSkusJSHandler::OnRefreshOrder(
+    v8::Global<v8::Promise::Resolver> promise_resolver,
+    v8::Isolate* isolate,
+    v8::Global<v8::Context> context_old,
+    const std::string& response) {
+  v8::HandleScope handle_scope(isolate);
+  v8::Local<v8::Context> context = context_old.Get(isolate);
+  v8::Context::Scope context_scope(context);
+  v8::MicrotasksScope microtasks(isolate,
+                                 v8::MicrotasksScope::kDoNotRunMicrotasks);
+
+  v8::Local<v8::Promise::Resolver> resolver = promise_resolver.Get(isolate);
+
+  base::JSONReader::ValueWithError value_with_error =
+      base::JSONReader::ReadAndReturnValueWithError(
+          response, base::JSONParserOptions::JSON_PARSE_RFC);
+  absl::optional<base::Value>& records_v = value_with_error.value;
+  if (!records_v) {
+    v8::Local<v8::String> result =
+        v8::String::NewFromUtf8(isolate, "Error parsing JSON response")
+            .ToLocalChecked();
+    ALLOW_UNUSED_LOCAL(resolver->Reject(context, result));
+    return;
+  }
+
+  const base::DictionaryValue* result_dict;
+  if (!records_v->GetAsDictionary(&result_dict)) {
+    v8::Local<v8::String> result =
+        v8::String::NewFromUtf8(isolate,
+                                "Error converting response to dictionary")
+            .ToLocalChecked();
+    ALLOW_UNUSED_LOCAL(resolver->Reject(context, result));
+    return;
+  }
+
+  v8::Local<v8::Value> local_result =
+      content::V8ValueConverter::Create()->ToV8Value(result_dict, context);
+  ALLOW_UNUSED_LOCAL(resolver->Resolve(context, local_result));
+}
+
+// window.brave.skus.fetch_order_credentials
+v8::Local<v8::Promise> BraveSkusJSHandler::FetchOrderCredentials(
```

Comment:
this needs to be converted to gin in a follow-up
see https://github.com/brave/brave-browser/issues/19909

Reply:
Do you have an example of using gin?

Reply:
see the link

Reply:
Did some renaming in https://github.com/brave/brave-core/pull/10847/commits/f9c99a4e9ae1cb7bc1ae3b57b7f51b1fe15e3788 to prepare for this change. Using [chrome/renderer/net/net_error_page_controller.h](https://source.chromium.org/chromium/chromium/src/+/main:chrome/renderer/net/net_error_page_controller.h)  and [chrome/renderer/net/net_error_helper.h](https://source.chromium.org/chromium/chromium/src/+/main:chrome/renderer/net/net_error_helper.h) as an example

Reply:
Created https://github.com/brave/brave-browser/issues/20437 as a follow up 👍